### PR TITLE
Support custom formatters for GFI responses

### PR DESCRIPTION
**Pull request by mingfeng**
Example changes needed in sample application:

```javascript
class TimeSeriesLatestPhotoYearFormatter {

    static identifier = "latestPhotoYear";

    enabled(data) {
        const layer = Oskari.getSandbox().getService('Oskari.mapframework.service.MapLayerService').findMapLayer(data.layerId);
        return layer.getAttributes().formatter === TimeSeriesLatestPhotoYearFormatter.identifier;
    }

    format(data) {
        const geojson = JSON.parse(data.content);
        const photoYearAttr = 'kuvausvuosi';
        const years = geojson.features.map(feature => parseInt(feature.properties[photoYearAttr], 10));
        return `<b>${photoYearAttr}:</b> ${Math.max(...years)}`;
    }
};


jQuery(document).ready(function() {
    function onSuccess () {
        var service = Oskari.getSandbox().getService('Oskari.map.DataProviderInfoService');
        if (service) {
            service.addGroup('osm.nominatim', 'Search', [
                { 'id': 'nominatim.attribution', 'name': 'The search functionality uses OpenStreetMap Nominatim, licensed under ODbL <a href="https://www.openstreetmap.org/copyright" target="_blank" ref="noopener">© OpenStreetMap contributors</a>' }
            ]);
        }
        const formatter = new TimeSeriesLatestPhotoYearFormatter();
        Oskari.getSandbox().findRegisteredModuleInstance('MainMapModuleGetInfoPlugin').addLayerFormatter(formatter);
    }
    function onError () {
        jQuery('#mapdiv').append('Unable to start');
    }
    Oskari.app.loadAppSetup(ajaxUrl + 'action_route=GetAppSetup', window.controlParams, onError, onSuccess);
});

```

image

### CODE REVIEW

File: bundles/mapping/mapmodule/plugin/getinfo/GetFeatureInfoFormatter.js [hunk 1]
```diff
@@ -125,6 +126,11 @@ Oskari.clazz.category('Oskari.mapframework.mapmodule.GetInfoPlugin', 'formatter'
             .forEach(frag => baseDiv.append(frag));
         return baseDiv;
     },
+
```
Comment (ZakarFin) [inline bundles/mapping/mapmodule/plugin/getinfo/GetFeatureInfoFormatter.js:129]: Add description here what a formatter is and check that it has enabled and format functions before adding it.

File: api/mapping/timeseries/bundle.md [hunk 2]
```diff
@@ -62,6 +62,21 @@ After registering the new type, TimeseriesLayerService will create delegates aut
...
+  "ui": "player|range|none",
+  "metadata": {
+    "layer": "<metadata layer id>",
+    "attribute": "<time attribute used to filter metadata layer>",
+    "toggleLevel": "<map toggle level at which the metadata layer will be shown>"
+  }
+}
+```
+
+The metadata is in use when `range` ui is selected
```
Comment (ZakarFin) [inline api/mapping/timeseries/bundle.md:78]: This is the configuration that can be added to a layer. The bundle state is something like:
```
{
    "time": "<time in some format or an array(?)"
}
```

Comment (ZakarFin) [reply api/mapping/timeseries/bundle.md:78]: And that state is only recognized by the "range" UI right?

Comment (mingfeng) [reply api/mapping/timeseries/bundle.md]: Currently yes

**Pull request merged**